pkg/controller/service/reconciler.go

```diff
@@ -106,13 +107,9 @@ func NewServiceReconciler(ctx context.Context, consoleClient client.Client, conf
 	if err != nil {
 		return nil, err
 	}
-	if err := CapabilitiesAPIVersions(discoveryClient); err != nil {
-		return nil, err
-	}
 
 	go func() {
-		//nolint:all
-		_ = wait.PollImmediateInfinite(time.Minute*5, func() (done bool, err error) {
+		_ = wait.PollUntilContextCancel(ctx, time.Minute*5, true, func(_ context.Context) (done bool, err error) {
```

Comment:
Does this start the polling immediately?  Just want to make sure there isn't a race.  Might even be worth callling `CapabilitiesApiVersions` immediately and swallowing errors to prevent that.

Reply:
It does start immediately, there is `true` arg passed that controls that. Anyway, since it is started in a separate routine, the code will continue without waiting. I can adjust that if needed and force calling function to wait.

Reply:
Yeah the only race I can foresee then is the goroutine happens to start after service reconcilers start, so running it synchronously first and discarding errors will at least fix that ordering problem